Guardian Druid-Optimization owlweaving opener/APL

Cathaklysme opened the pull request:
Changing the opener of Owlweaving to get Solar eclipse when using convoke with the night fae covenant
Getting rid the bear form condition on barkskin
Optimizing the APL for Owlweaving

Cathaklysme committed:
Guardian Druid-Optimization owlweaving opener/APL

Changing the opener of Owlweaving to get Solar eclipse when using convoke with the night fae covenant
Getting rid the bear form condition on barkskin
Optimizing the APL for Owlweaving

Cathaklysme committed:
Update sc_druid.cpp

Cathaklysme committed:
Update sc_druid.cpp

Cathaklysme committed:
Update sc_druid.cpp

Cathaklysme committed:
Update sc_druid.cpp

Cleaning up the shifting conditions for moonkin, and removing the starfire opener line which should fix the issue with precombat

Cathaklysme committed:
Update sc_druid.cpp

Melekus commented on engine/class_modules/sc_druid.cpp:
```diff
@@ -8690,7 +8689,7 @@ void druid_t::apl_guardian()
   bear->add_action( "incarnation,if=(buff.ravenous_frenzy.up|!covenant.venthyr)" );
   bear->add_action( "empower_bond,if=(!druid.catweave_bear&!druid.owlweave_bear)|active_enemies>=2" );
-  bear->add_action( "barkskin,if=(talent.brambles.enabled)&(buff.bear_form.up)" );
+  bear->add_action( "barkskin,if=(talent.brambles.enabled)" );
```
```suggestion
  bear->add_action( "barkskin,if=talent.brambles.enabled" );
```

Melekus reviewed (approved):
Some of the longer lines like could use commenting (for example the owlweave run action list).
Looks good 👍

Cathaklysme committed:
Update sc_druid.cpp

Adding comments and cleaning up useless condition from the catweaving apl related to the Oath legendary

Cathaklysme committed:
Update engine/class_modules/sc_druid.cpp

Co-authored-by: Antoine C. <antoine.colin1@hotmail.fr>

Cathaklysme committed:
Update sc_druid.cpp

Cathaklysme committed:
Update sc_druid.cpp

Cathaklysme committed:
Update sc_druid.cpp
